web/src/message_list_view.js

```diff
@@ -501,51 +511,57 @@ export class MessageListView {
                 // Highlight messages having personal mentions only in DMs and subscribed streams.
                 if (
-                    message_container.msg.type === "private" ||
-                    stream_data.is_user_subscribed(
-                        message_container.msg.stream_id,
-                        people.my_current_user_id(),
-                    )
+                    message.type === "private" ||
+                    stream_data.is_user_subscribed(message.stream_id, people.my_current_user_id())
                 ) {
-                    message_container.mention_classname = "direct_mention";
+                    mention_classname = "direct_mention";
                 }
             } else {
-                message_container.mention_classname = "group_mention";
+                mention_classname = "group_mention";
             }
         } else {
             // If there are no mentions, the classname might need to be updated (i.e.
             // removed) to reflect this.
-            message_container.mention_classname = null;
+            mention_classname = null;
         }
-        message_container.include_sender = message_container.include_sender && !is_hidden;
+        let include_sender = existing_include_sender && !is_hidden;
         if (is_revealed) {
             // If the message is to be revealed, we show the sender anyways, because the
             // the first message in the group (which would hold the sender) can still be
             // hidden.
-            message_container.include_sender = true;
+            include_sender = true;
         }
 
-        message_container.sender_is_bot = people.sender_is_bot(message_container.msg);
-        message_container.sender_is_guest = people.sender_is_guest(message_container.msg);
-        message_container.should_add_guest_indicator_for_sender =
-            people.should_add_guest_user_indicator(message_container.msg.sender_id);
+        const sender_is_bot = people.sender_is_bot(message);
+        const sender_is_guest = people.sender_is_guest(message);
+        const should_add_guest_indicator_for_sender = people.should_add_guest_user_indicator(
+            message.sender_id,
+        );
 
-        message_container.small_avatar_url = people.small_avatar_url(message_container.msg);
-        if (message_container.msg.stream_id) {
-            message_container.background_color = stream_data.get_color(
-                message_container.msg.stream_id,
-            );
+        const small_avatar_url = people.small_avatar_url(message);
+        let background_color;
+        if (message.type === "stream") {
+            background_color = stream_data.get_color(message.stream_id);
         }
 
-        Object.assign(
-            message_container,
-            this._maybe_get_me_message(message_container.is_hidden, message_container.msg),
-        );
-        // Once all other variables are updated
```

Comment:
I couldn't find any reason this had to happen after other variables were updated.

Reply:
This comment was added in 1f76374e32a547bfcf10096422ad89fc3d585c65; I think it's because the edit placement variables were a function of details like whether it's a me message.

ea2b659314459b571ac0fa627fd2617c2cce4f3c reworked this, and looks like it failed to delete some code around `edited_alongside_sender` in `this._add_msg_edited_vars`, so we should clean that up soon.